### PR TITLE
OData join clause mapping table

### PR DESCRIPTION
**Pull request by BorisNen**
Related to https://github.com/SAP/xsk/issues/1410

Changelog:
1. Added a mapping table to the ODataAssociationEndDefinition.
2. OData2ODataMTransformer will now create a json with a mapping table property.
3. SQLJoinClause will now check for the existence of a mapping table and construct a join statement with it.
4. Added 2 unit tests for the mapping table in the transformer and SQL builder in the SQLJoinClause.

### CODE REVIEW

File: modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/clause/SQLJoinClause.java [hunk 1]
```diff
@@ -63,29 +63,77 @@ public String evaluate(SQLContext context) throws EdmException {
...
             return "";
         }
-        boolean caseSensitive = Boolean.parseBoolean(Configuration.get("DIRIGIBLE_DATABASE_NAMES_CASE_SENSITIVE", "false"));
 
         StringBuilder join = new StringBuilder();
+
+        boolean hasFirstJsonMappingTable = query.hasSQLMappingTablePresent(start, target);
+        boolean hasSecondJsonMappingTable = query.hasSQLMappingTablePresent(target, start);
+
+        if(hasFirstJsonMappingTable && hasSecondJsonMappingTable) {
```
Comment (MarinHadzhiev) [inline modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/clause/SQLJoinClause.java:72]: What will happen if mapping table from the two entities is different? Can we add check for that?

File: modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/builder/AbstractQueryBuilder.java [hunk 2]
```diff
@@ -232,4 +268,23 @@ public String grantTableAliasForStructuralTypeInQuery(final EdmStructuralType en
...
+                    // Alias is already contained in the map
+                    return getTableAliasForManyToManyMappingTable(target);
+            }
+            String alias = "MT" + tableAliasesForManyToManyMappingTablesInQuery.size();
+            LOG.debug("Grant Alias '" + alias + "' for " + manyToManyMappingTable);
+            // Add alias to map
+            tableAliasesForManyToManyMappingTablesInQuery.put("MT" +
+                    tableAliasesForManyToManyMappingTablesInQuery.size(), manyToManyMappingTable);
+            return alias;
+        } catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (BorisNen) [reply modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/builder/AbstractQueryBuilder.java]: Resolved in [f66d649](https://github.com/eclipse/dirigible/pull/1685/commits/f66d649aeb8e56d59d0b4a748b56a5d89dd8c1fd)

File: modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/clause/SQLJoinClause.java [hunk 3]
```diff
@@ -63,29 +63,77 @@ public String evaluate(SQLContext context) throws EdmException {
...
+            join.append(" ");
+
+            List<String> secondJoinColumns = query.getSQLMappingTableJoinColumn(start, target);
+            String secondJoinLeftTableAlias = query.getSQLTableAliasForManyToManyMappingTable(firstJoinRightTable);
+            String secondJoinRightTable = query.getSQLTableName(start);
+            String secondJoinRightTableAlias = query.getSQLTableAlias(start);
+            List<String> secondJoinTargetKeys = getTargetJoinKeyForEntityType(start, target);
+
+            buildJoinClause(secondJoinColumns, secondJoinLeftTableAlias, secondJoinRightTable, secondJoinRightTableAlias,
+                    secondJoinTargetKeys, join);
```
[the remark this answers is not in the document]
Comment (BorisNen) [reply modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/clause/SQLJoinClause.java]: Resolved in [f66d649](https://github.com/eclipse/dirigible/pull/1685/commits/f66d649aeb8e56d59d0b4a748b56a5d89dd8c1fd)

File: modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/clause/SQLJoinClause.java (hunk 1 above)
Comment (BorisNen) [reply modules/odata/odata-core/src/main/java/org/eclipse/dirigible/engine/odata2/sql/clause/SQLJoinClause.java:72]: Resolved in [f66d649](https://github.com/eclipse/dirigible/pull/1685/commits/f66d649aeb8e56d59d0b4a748b56a5d89dd8c1fd)

**Review by MarinHadzhiev — APPROVED** [review]

**Pull request merged**